syntax = "proto3";

package apollo.fuelation;

import "fueling/learning/autotuner/proto/dynamic_model_info.proto";
import "fueling/learning/autotuner/proto/git_info.proto";

message Parameter {
  string parameter_name = 1;
  double min = 2;
  double max = 3;
  bool is_repeated = 4;
}

message Flag {
  string flag_name = 1;
  bool enable = 2;
}

message TunerParameters {
  string default_conf_proto = 1;
  string default_conf_filename = 2;
  Utility utility = 3;
  int32 n_iter = 4;  // Number of iterations
  int32 init_points_1D = 5;  //Number of initial points per parameter
  bool opt_max = 6;  // If True, explore the maximum; or, explore the minimum
  repeated Flag flag = 7;
  repeated Parameter parameter = 8;
}

message Scenarios {
  repeated uint32 id = 3;
}

message TunerConfigs {
  GitInfo git_info = 1;
  TunerParameters tuner_parameters = 2;
  Scenarios scenarios = 3;
  DynamicModel dynamic_model = 4;
}

message Utility {
  string utility_name =
      1;  // Utility function can be set as "ucb", "ei" or "poi"
  double kappa = 2;
  double xi = 3;
}
